### PR TITLE
Create sync-platform-branch action

### PR DESCRIPTION
**Pull request by francinelucca**
Creates github action to automatically sync carbon-platform branch with main

#### Changelog

**New**

- .github/workflows/sync-platform-branch: github action

**Test**
Make sure I'm using options correctly as specified by: https://github.com/TreTuna/sync-branches#sync-branches

### CODE REVIEW

**Review by tw15egan — APPROVED** [review]

File: .github/workflows/sync-platform-branch [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  # Enable running this workflow manually from the Actions tab
+  workflow_dispatch: # Allow for running this manually.
+
+jobs:
+  sync-branches:
+    runs-on: ubuntu-latest
+    name: Syncing branches
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (tay1orjones) [inline .github/workflows/sync-platform-branch:24]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/sync-platform-branch [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      - name: Set up Node
+        uses: actions/setup-node@v1
```
Comment (tay1orjones) [inline .github/workflows/sync-platform-branch:26]:
```suggestion
        uses: actions/setup-node@v3
```

File: .github/workflows/sync-platform-branch [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        with:
+          node-version: 12
```
Comment (tay1orjones) [inline .github/workflows/sync-platform-branch:28]: Actions using node 12 are deprecated
```suggestion
          node-version: 16
```

**Review by tay1orjones — APPROVED** [review]
Some of the steps can use newer versions of the actions, otherwise looks good!

**Pull request merged**